### PR TITLE
Fix rare issue where transactions can be underpriced in EIP-1559 mode

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/logger/logger.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func init() {
 //
 // Levels
 //  - Fatal: Logs and then calls os.Exit(1). Be careful about using this since it does NOT unwind the stack and may exit uncleanly.
-//  - Panic: Unrecoverable error. Example: invariant violation, programmer error
+//  - Panic: Unrecoverable error. Example: AssumptionViolation, programmer error
```
Comment (jmank88) [inline core/logger/logger.go:46]: Did we come to a conclusion about these? Why replace a well known idiomatic term here?

Comment (samsondav) [reply core/logger/logger.go:46]: I thought the logging conclusion was to log AssumptionViolation and document that everywhere?

Comment (samsondav) [reply core/logger/logger.go:46]: I'm just going to remove that commit from this PR to avoid any unrelated controversy

**Review by prashantkumar1982 — DISMISSED** [review]

File: core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go [hunk 2]
```diff
@@ -1373,8 +1373,14 @@ func TestEthBroadcaster_ProcessUnstartedEthTxs_Errors(t *testing.T) {
...
 		// to zero (even though that should not be possible due to config
 		// validation)
+		oldGasBumpWei := evmcfg.EvmGasBumpWei()
+		oldGasBumpPercent := evmcfg.EvmGasBumpPercent()
 		cfg.Overrides.GlobalEvmGasBumpWei = big.NewInt(0)
 		cfg.Overrides.GlobalEvmGasBumpPercent = null.IntFrom(0)
+		defer func() {
+			cfg.Overrides.GlobalEvmGasBumpWei = oldGasBumpWei
+			cfg.Overrides.GlobalEvmGasBumpPercent = null.IntFrom(int64(oldGasBumpPercent))
+		}()
```
Comment (jmank88) [inline core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go:1383]: Is this resetting shared state from the parent test? Worth a comment at least?
Would `t.Cleanup` be more or less reliable? :thinking: maybe only different in irrelevant situations anyways...

**Review by jmank88 — APPROVED** [review]

File: core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go (hunk 2 above)
Comment (samsondav) [reply core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go:1383]: Yes it's resetting state. `t.Cleanup` might be ambiguous, is it this `t` or the parent one.

Comment (jmank88) [reply core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go:1383]: IMHO it reads pretty clearly, but actually from a higher level, is it even safe to do any mutations on cfg from subtests? Once the cfg is in use, these are all racy, aren't they?

Comment (prashantkumar1982) [reply core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go:1383]: Could subset tests run in parallel?

**Review by prashantkumar1982 — APPROVED** [review]

File: core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go (hunk 2 above)
Comment (jmank88) [reply core/chains/evm/bulletprooftxmanager/eth_broadcaster_test.go:1383]: Only if explicitly marked as Parallel, but they all use long running concurrent components from the _parent_ test that could race to read these overrides.

**Pull request merged**